### DISCUSSION TITLE
Quire Preview- Local Host 404 Page Not Found Error

### DISCUSSION THREAD

**Discussion by aebancroft** [thegetty/quire#355] [Get Help]
Hello!

I was able to view my new Quire project in preview a couple days ago, but now subsequent updates haven't displayed and I'm unable to preview the project from http://localhost:1313/paintings/nmunz-bellotto. I get a "404 page not found" error and don't believe I have errors from the command prompt, at least not through regular Quire Preview. I've attached the verbose output, but admit I'm a little stumped by it. 

I just tried reverting the commits I've made since the last time I was able to preview the project - no luck unfortunately.

Any suggestions you have are greatly appreciated!

Thanks!

Alex

[Quire Verbose Output 2022 06 15.txt](https://github.com/thegetty/quire/files/8912166/Quire.Verbose.Output.2022.06.15.txt)

**Comment by Erin-Cecele** [comment]
Hi @aebancroft can you run the command `node --version` and tell us what version of Node.js you are running on?

**Reply by aebancroft** [reply]
Hi, @Erin-Cecele ! Node = v12.18.1. 

I was trying to crank out one more project before switching over to 11ty as well. Let me know if it would be better to do that now.

**Reply by geealbers** [reply]
@aebancroft, are you on a Mac, and recently updated to Monterey? If so, you'll need to install Python 2. Follow this link: https://www.python.org/downloads/release/python-2718/, click the “macOS 64-bit installer,” and follow the install prompts. And then before trying preview again, run `quire install` once. I think that all might clear it up. 🤞

**Reply by aebancroft** [reply]
Thanks, @geealbers. I'm on a PC though.

**Reply by geealbers** [reply]
Ah, right. Can you run `quire --version` and let us know the output?

**Reply by aebancroft** [reply]
v0.19.2

**Reply by aebancroft** [reply]
Hi - I'm not sure if this helps troubleshoot, but I ran Quire preview again today and am finding that older, published essays are also not appearing with the latest updates; as if preview is looking at an old version of the repo. I.e. http://localhost:1313/paintings/sbondini-pino/ vs.  https://stage.museum-essays.getty.edu/paintings/sbondini-pino/.

**Reply by aebancroft** [reply]
Hi again - I deleted my local copy of the repo, cloned again, and re-ran Quire preview. Now I'm getting this:

`C:\Users\ABancroft\museum-essays>quire preview
Launching preview server
i Navigate to http://localhost:1313 to see your changes.
i Press Control+C or type "quire stop" to stop the preview.
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\webpack.cmd ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn node_modules\\.bin\\webpack.cmd',
  path: 'node_modules\\.bin\\webpack.cmd',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack\\webpack.config.dev.js'
  ]
}`

**Reply by Erin-Cecele** [reply]
Hi @aebancroft trying running 'quire install' before 'quire preview'.

**Reply by aebancroft** [reply]
Thanks, @Erin-Cecele. Ran Quire install and this was the output:

`C:\Users\ABancroft\museum-essays>quire install quire preview
[            ......] | finalize:lodash.get: sill finalize C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\no
> node-sass@4.11.0 install C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.11.0/win32-x64-72_binding.node
Cannot download "https://github.com/sass/node-sass/releases/download/v4.11.0/win32-x64-72_binding.node":

HTTP error 404 Not Found

Hint: If github.com is not accessible in your location
      try setting a proxy via HTTP_PROXY, e.g.

      export HTTP_PROXY=http://example.com:1234

or configure npm proxy via

      npm config set proxy http://example.com:8080

> gifsicle@4.0.1 postinstall C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\gifsicle
> node lib/install.js

  √ gifsicle pre-build test passed successfully

> jpegtran-bin@4.0.0 postinstall C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\jpegtran-bin
> node lib/install.js

  √ jpegtran pre-build test passed successfully

> optipng-bin@5.1.0 postinstall C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\optipng-bin
> node lib/install.js

  √ optipng pre-build test passed successfully

> node-sass@4.11.0 postinstall C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\node-sass
> node scripts/build.js

Building: C:\Program Files\nodejs\node.exe C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   'C:\\Program Files\\nodejs\\node.exe',
gyp verb cli   'C:\\Users\\ABancroft\\museum-essays\\themes\\quire-starter-theme\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library='
gyp verb cli ]
gyp info using node-gyp@3.8.0
gyp info using node@12.18.1 | win32 | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:167:21)
gyp verb `which` failed  python2 Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:167:21) {
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python C:\Python38\python.EXE
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Python38\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                       ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:303:12)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
gyp ERR! System Windows_NT 10.0.19044
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\ABancroft\\museum-essays\\themes\\quire-starter-theme\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd C:\Users\ABancroft\museum-essays\themes\quire-starter-theme\node_modules\node-sass
gyp ERR! node -v v12.18.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
Build failed with error code: 1
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-sass@4.11.0 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-sass@4.11.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ABancroft\AppData\Roaming\npm-cache\_logs\2022-06-16T22_00_33_014Z-debug.log
√ Theme dependencies successfully installed.
√ Run quire preview to view changes locally.`

Unfortunately the preview issue is still present. Any other ideas?

**Reply by geealbers** [reply]
@aebancroft On that last suggestion, you'd need to run `quire install` by itself first, and then when that is finished run `quire preview --verbose`. I'm not sure if that will fix it though, and we're running into a bit of a wall on this one as I'm going to be out for a week starting tomorrow, and would need more time than I have before then to do testing on a PC.

Hoping @workergnome might be able to step in a suggest a path forward? Perhaps Getty Digital can look at it for you more formally?

**Reply by aebancroft** [reply]
Thanks, Greg and Erin! I did run quire install by itself and then quire preview --verbose. Still getting the same output as above. 

@workergnome if you have any suggestions please let me know! Or if there's someone else in Getty Digital I shoudl contact. 

Thank you!

**Reply by aebancroft** [reply]
@geealbers  and @Erin-Cecele thanks for your help. David N. and Adam B. helped narrow this down to an issue with how the essays are published to our website; not with Quire itself. We relied on Netlify which is no longer used. Just wanted to let you know this one is resolved! Thanks!

**Reply by Erin-Cecele** [reply]
Thank you for the update @aebancroft!
